x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/SnapshotLifecyclePolicy.java

```diff
@@ -103,8 +113,29 @@ public ValidationException validate() {
         return null;
     }
 
+    /**
+     * Since snapshots need to be uniquely named, this method will resolve any date math used in
+     * the provided name, as well as appending a unique identifier so expressions that may overlap
+     * still result in unique snapshot names.
+     */
+    public String generateSnapshotName(Context context) {
+        List<String> candidates = DATE_MATH_RESOLVER.resolve(context, Collections.singletonList(this.name));
+        if (candidates.size() != 1) {
+            throw new IllegalStateException("resolving snapshot name " + this.name + " generated more than one candidate: " + candidates);
+        }
+        // TODO: we are breaking the rules of UUIDs by lowercasing this here, find an alternative (snapshot names must be lowercase)
```

Comment:
So we break rules because we are using base64 UUIDs, so the problem is that these UUIDs which are all different:

```
kwNO8wpLQFaSwa5RSbvDvg
KwNO8wplqFaSwa5RSbvdvg
kwNo8wpLQfASwA5rSbvDvg
```

Are all going to result in the same snapshot name (assuming no date math), even though they're different UUIDs, since they're lower-cased and case is important for base64

Reply:
Right, I forgot we use base64-encoded UUIDs instead of hex-encoded UUIDs. Even though it says base64 right there in the method call. Anyway, yeah, we should probably find something better long term but this is fine for now.